pyServer/AzureDiskInspectService.py

```diff
@@ -472,6 +472,16 @@ def do_POST(self):
                     self.telemetryLogger.info('WARNING: Received timeout override is invalid: {0}. Default will be used.'.format(timeoutInMinsStr))
 
             timeoutInMins = int(timeoutInMinsStr)
+
+            self.telemetryLogger.info('Received timeout value: ' + str(timeoutInMins)+ ' min(s)')
+
+            if timeoutInMins < 20:
+                zipFileHandlingOverhead = 1 # If customized timeout is low, inspection result will likely be small. Zip archive creation will be faster.
+            else:
+                zipFileHandlingOverhead = 4  # if customized timeout is high, inspection result will likely be large and zip archiving time will take longer.
+
+            self.telemetryLogger.info('Trimming timeout value by ' + str(zipFileHandlingOverhead)+ ' min(s) to reserve overhead time for inspection file archiving.')
+            timeoutInMins -= zipFileHandlingOverhead
```

Comment:
Do we need to document or advise the caller of this in the log?  I'd expect using a timeout less than what was asked for would be considered unexpected

Reply:
So the trimmed time out value in mins is already logged in the response header. We could add another header to show how much was trimmed. 

But I'm thinking the only client using the extended timeout feature right now is ADS, which is the main targeted beneficiary for this fix. ADS doesn't really have a way to interpret additional logging if we told them some time was trimmed for file handling overhead. For other scenarios, a trim of one minute should be pretty minimal and EG wouldn't know to process that information either. 

Thoughts?

Reply:
Trimmed timeout value == KeepAliveThread-Timeout-In-Mins?

Reply:
Yes, we log the trimmed timeout value in the "KeepAliveThread-Timeout-In-Mins" header.